### ISSUE TITLE
Some XRay tests in clang/test/Driver were unsupported on x86_64

### ISSUE THREAD

**Issue by SixWeining** [llvm/llvm-project#64028]
```
$ arch
x86_64
```

```
$ ./_build/bin/llvm-lit clang/test/Driver/XRay/
llvm-lit: /home/luweining/upstreams/llvm-project/llvm/utils/lit/lit/llvm/config.py:487: note: using clang: /home/luweining/upstreams/llvm-project/_build/bin/clang
-- Testing: 6 tests, 6 workers --
UNSUPPORTED: Clang :: Driver/XRay/xray-shared-noxray.cpp (1 of 6)
UNSUPPORTED: Clang :: Driver/XRay/xray-ignore-loops-flags.cpp (2 of 6)
UNSUPPORTED: Clang :: Driver/XRay/xray-instrumentation-bundles-flags.cpp (3 of 6)
UNSUPPORTED: Clang :: Driver/XRay/xray-mode-flags.cpp (4 of 6)
PASS: Clang :: Driver/XRay/xray-nolinkdeps.cpp (5 of 6)
PASS: Clang :: Driver/XRay/xray-instrument.c (6 of 6)

Testing Time: 0.97s
  Unsupported: 4
  Passed     : 2
```

**Comment by SixWeining** [comment]
These tests are `PASS` in previous version, e.g. llvm15. I'm trying to do `git bisect` to find which commit causes the issue.

**Comment by SixWeining** [comment]
`git bisect` finds 016785d9316d8c5abc5fdf3cdb86479095bbb677. I tried to fix it and request review in [D156089](https://reviews.llvm.org/D156089).
